plugins/usbdmx/EuroliteProFactory.cpp

```diff
@@ -32,30 +32,80 @@ namespace usbdmx {
 
 using ola::usb::LibUsbAdaptor;
 
+// "Eurolite USB-DMX512-PRO"
 const char EuroliteProFactory::EXPECTED_MANUFACTURER[] = "Eurolite";
 const char EuroliteProFactory::EXPECTED_PRODUCT[] = "Eurolite DMX512 Pro";
 const uint16_t EuroliteProFactory::PRODUCT_ID = 0xfa63;
 const uint16_t EuroliteProFactory::VENDOR_ID = 0x04d8;
 
+// "Eurolite USB-DMX512-PRO MK2" (successor device introduced in late 2016)
+const char EuroliteProFactory::EXPECTED_MANUFACTURER_MK2[] = "FTDI";
+const char EuroliteProFactory::EXPECTED_PRODUCT_MK2[] = "FT232R USB UART";
+const uint16_t EuroliteProFactory::PRODUCT_ID_MK2 = 0x6001;
+const uint16_t EuroliteProFactory::VENDOR_ID_MK2 = 0x0403;
+
+const char EuroliteProFactory::ENABLE_EUROLITE_MK2_KEY[] =
+  "enable_eurolite_mk2";
+
+EuroliteProFactory::EuroliteProFactory(ola::usb::LibUsbAdaptor *adaptor,
+                                       Preferences *preferences)
+  : BaseWidgetFactory<class EurolitePro>("EuroliteProFactory"),
+    m_adaptor(adaptor),
+    m_enable_eurolite_mk2(IsEuroliteMk2Enabled(preferences)) {
+}
+
+bool EuroliteProFactory::IsEuroliteMk2Enabled(Preferences *preferences) {
+  bool enabled;
+  if (!StringToBool(preferences->GetValue(ENABLE_EUROLITE_MK2_KEY),
+                    &enabled)) {
+    enabled = false;
+  }
+  return enabled;
+}
+
 bool EuroliteProFactory::DeviceAdded(
     WidgetObserver *observer,
     libusb_device *usb_device,
     const struct libusb_device_descriptor &descriptor) {
-  if (descriptor.idVendor != VENDOR_ID || descriptor.idProduct != PRODUCT_ID) {
-    return false;
-  }
+  bool isMK2 = false;
 
-  OLA_INFO << "Found a new EurolitePro device";
-  LibUsbAdaptor::DeviceInformation info;
-  if (!m_adaptor->GetDeviceInfo(usb_device, descriptor, &info)) {
-    return false;
-  }
+  // Eurolite USB-DMX512-PRO?
+  if (descriptor.idVendor == VENDOR_ID && descriptor.idProduct == PRODUCT_ID) {
+    OLA_INFO << "Found a new Eurolite USB-DMX512-PRO device";
+    LibUsbAdaptor::DeviceInformation info;
+    if (!m_adaptor->GetDeviceInfo(usb_device, descriptor, &info)) {
+      return false;
+    }
 
-  if (!m_adaptor->CheckManufacturer(EXPECTED_MANUFACTURER, info)) {
-    return false;
-  }
+    if (!m_adaptor->CheckManufacturer(EXPECTED_MANUFACTURER, info)) {
+      return false;
+    }
+
+    if (!m_adaptor->CheckProduct(EXPECTED_PRODUCT, info)) {
+      return false;
+    }
 
-  if (!m_adaptor->CheckProduct(EXPECTED_PRODUCT, info)) {
+  // Eurolite USB-DMX512-PRO MK2?
+  } else if (m_enable_eurolite_mk2 &&
+             descriptor.idVendor == VENDOR_ID_MK2 &&
+             descriptor.idProduct == PRODUCT_ID_MK2) {
+    OLA_INFO << "Found a new Eurolite USB-DMX512-PRO MK2 device";
```

Comment:
I suspect it might be handy to restructure this to prompt people via a log line if it finds a potential Mk2 and they're disabled, rather than them just being confused why it's not working.

Reply:
I've done that now, and tested it by disabling mk2, and get the message that a possible mk2 is found.